frontend/app/blueprints/public/static/js/basket.js

```diff
@@ -0,0 +1,136 @@
+// functions relating to the sample basket and navbar
+async function openGrapeTree({newick, sampleIds, clusterMethod}) {
```

Comment:
What do you think about using JSDoc here?

```
 /**
  * @param {Object} obj
  * @param {NewickTree} obj.newickTree 
  * @param {string[]} obj.sampleIds 
  * @param {"method1"|"method2"} obj.clusterMethod
  */
async function openGrapeTree({newick, sampleIds, clusterMethod}) {
```

Just food for thought. This is a long term decision.